internal/mining/bgblktmplgenerator.go

```diff
@@ -0,0 +1,1439 @@
+// Copyright (c) 2014-2016 The btcsuite developers
```

Comment:
This should probably be changed to just be the following:

```Go
// Copyright (c) 2019-2020 The Decred developers
// Use of this source code is governed by an ISC
// license that can be found in the LICENSE file.
```

This is all new code that I wrote specifically for Decred.  There is nothing from btcsuite nor prior to 2019 in here.